### ISSUE TITLE
support aws auth

### ISSUE THREAD

**Issue by dseynaev** [vimc/vaultr#28]
Hi there, thanks for this wonderful package.

I was missing AWS auth login support and started implementing it in a fork.
Is this something that could be in scope for `vaultr`? If desired, I can cleanup my code
and start working on a PR.

Reference: https://www.vaultproject.io/docs/auth/aws

**Comment by richfitz** [comment]
Yes, this would be absolutely in scope, and a PR would be great.  If you can think of a sensible way of testing it, it'd be ideal - the implementation you have looks straightforward enough, and we've found `mockery` useful for much of this sort of thing, though I don't think we use it much in `vaultr` yet

**Comment by dseynaev** [comment]
I'll look into how to test it.

For signing the request I used `aws.signature`. How do you want me to handle that extra dependency? Is it okay to add it to `Suggests`?

**Comment by richfitz** [comment]
Suggests seems good to me, thanks

**Comment by richfitz** [comment]
Hi @salarali - nothing from our end as we do not use AWS as an auth provider. However, we would still be open to a PR if anyone would like to contribute support

**Comment by dseynaev** [comment]
I have this implemented in a fork and been using it for a while now without major issues. I can try to clean it up and add tests for a PR.

**Comment by richfitz** [comment]
Thanks @dseynaev - your fork looks pretty tidy so far as I can see, so with some tests that should be pretty easy to fold in

**Comment by vnellore** [comment]
Hello, I am interested the AWS authentication feature for this package. Any idea when this will be merged? Thanks!

**Comment by richfitz** [comment]
No progress on this - someone would ideally write some tests (we don't use AWS auth internally). The code is here https://github.com/vimc/vaultr/compare/master...dseynaev:vaultr:master and looks nice and tidy, but as I don't use AWS auth I can't really much with it without tests

**Comment by vnellore** [comment]
Thanks! @richfitz !
@dseynaev Please let me know if any help is needed. This would be a great feature addition for this package. Thank you!

**Comment by fh-mthomson** [comment]
@dseynaev @vnellore I'm curious if either of you are still interested in pushing this forward? 

Thank you for your interest and initial efforts - it would be amazing to be able to consume via `vaultr`!
